src/Components/Blueprints/BlueprintActionsMenu.tsx

```diff
@@ -60,11 +71,36 @@ export const BlueprintActionsMenu: React.FunctionComponent<
         >
           Edit details
         </DropdownItem>
-        <DropdownItem>Download blueprint (.json)</DropdownItem>
+        <DropdownItem
+          onClick={() => {
+            trigger({ id: selectedBlueprintId || '' });
+          }}
+        >
+          Download blueprint (.json) <BetaLabel />
+        </DropdownItem>
         <DropdownItem onClick={() => setShowDeleteModal(true)}>
           Delete blueprint
         </DropdownItem>
       </DropdownList>
     </Dropdown>
   );
 };
+
+async function handleExportBlueprint(
+  blueprintName: string | undefined,
+  blueprint: BlueprintResponse | undefined
```

Comment:
Can we refactor this so that blueprintName is only `string` and blueprint is only `BlueprintResponse`? Let's deal with the undefined case at the call site.